subiquity/models/subiquity.py

```diff
@@ -378,6 +378,8 @@ def _cloud_init_config(self):
                 user_info["ssh_authorized_keys"] = self.ssh.authorized_keys
             config["users"] = [user_info]
         else:
+            if self.userdata is None:
+                config["users"] = []
             if self.ssh.authorized_keys:
                 config["ssh_authorized_keys"] = self.ssh.authorized_keys
```

Comment:
To hit that you need all of the following
* desktop
* no `identity`
* no `user-data`
* a subiquity `ssh.authorized-keys`

at this point, the authorized_keys will be populated for the root.  If root login is disabled in ssh (default!), it will be subject to https://github.com/canonical/cloud-init/blob/621c522ccf2897ff4d19a8ed332c187516eccf9b/cloudinit/ssh_util.py#L60

I don't think it hurts anything.  I'm adding a log.warning but not interfering with sending the authorized-keys along.